CRM/Mailing/BAO/Mailing.php

```diff
@@ -357,22 +357,40 @@ public static function getRecipients($mailingID) {
       $query = $query->where($aclWhere);
     }
 
+    // TODO: read the directory from the MySQL server
+    $tmpfname = self::tmpfile_for_select_into("/var/lib/mysql-files",$mailingID);
+
     // this mean if dedupe_email AND the mysql 5.7 supports ONLY_FULL_GROUP_BY mode then as
     //  SELECT must contain 'email' column as its used in GROUP BY, so in order to resolve This
     //  here the whole SQL code is wrapped up in FROM table i and not selecting email column for INSERT
     if ($key = array_search('e.email', $selectClause)) {
       unset($selectClause[$key]);
       $sql = $query->toSQL();
-      CRM_Utils_SQL_Select::from("( $sql ) AS i ")
+      $sql = CRM_Utils_SQL_Select::from("( $sql ) AS i ")
         ->select($selectClause)
-        ->insertInto('civicrm_mailing_recipients', ['mailing_id', 'contact_id', $entityColumn])
         ->param('#mailingID', $mailingID)
-        ->execute();
-    }
-    else {
-      $query->insertInto('civicrm_mailing_recipients', ['mailing_id', 'contact_id', $entityColumn])
+        ->toSQL();
+
+      // Error check?
+      $dao = CRM_Core_DAO::executeQuery("$sql INTO OUTFILE '$tmpfname'");
+      $dao = CRM_Core_DAO::executeQuery(
+        "LOAD DATA INFILE '$tmpfname' INTO TABLE civicrm_mailing_recipients" .
+          " (mailing_id, contact_id, $entityColumn)"
+      );
+    } else {
+      $sql = CRM_Utils_SQL_Select::from("( $sql ) AS i ")
+        ->select($selectClause)
         ->param('#mailingID', $mailingID)
-        ->execute();
+        ->toSQL();
+     
+      // Error check?
+      $dao = CRM_Core_DAO::executeQuery("SELECT $sql INTO OUTFILE $tmpfname");
+
+      // TODO: LOAD DATA FROM FILE
+      $dao = CRM_Core_DAO::executeQuery(
+        "LOAD DATA INFILE '$tmpfname' INTO civicrm_mailing_recipients" .
+          " (mailing_id, contact_id, $entityColumn)"
+      );
```

Comment:
Ah but we can't ! it's on the db server !

We'll need to add a cronjob to clean up the files on the db server.

Reply:
Ah right of course! :facepalm: But we have to, right? Otherwise the files are going to pile up. Maybe that's why I've never heard of this suggestion :)